File: 06Adapter/02TwoWayAdapter/Aircraft.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _02TwoWayAdapter {
    // Target
    public sealed class Aircraft : IAircraft {
        int height;
        bool airborne;
        public Aircraft() {
            height = 0;
            airborne = false;
        }
        public void TakeOff() {
            Console.WriteLine("Aircraft engine takeoff");
            airborne = true;
            height = 200; // Meters
        }
        public bool Airborne
        {
            get { return airborne; }
        }
        public int Height
        {
            get { return height; }
        }
    }
}

File: 06Adapter/Students/2019/AliaksandrNovik/src/main/java/twoWayAdapter/Aircraft.java
package twoWayAdapter;

public class Aircraft implements IAircraft {

	private boolean airborne;
	private int height;

	public Aircraft() {
		airborne = false;
		height = 0;
	}

	@Override
	public void takeOff() {
		System.out.println("Aircraft engine takeoff");
		airborne = true;
		height = 200;
	}

	@Override
	public boolean airborne() {
		return airborne;
	}

	@Override
	public int height() {
		return height;
	}
}